spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -66,6 +66,23 @@ def team_id
 
     # Set a new team ID which will be used from now on
     def team_id=(team_id)
+      # First, we verify the team actually exists, because otherwise iTC would return the
+      # following confusing error message
+      #
+      #     invalid content provider id
+      #
+      available_teams = teams.collect do |team|
+        team.fetch("contentProvider", {}).fetch("contentProviderId", nil)
+      end
+
+      result = available_teams.find do |available_team_id|
+        team_id.to_s == available_team_id.to_s
```

Comment:
Does it make sense to stop iterating once we find the `available_team_id` that matches?